src/scheduler/node_partition.cpp

```diff
@@ -1403,3 +1405,60 @@ update_all_nodepart(status *policy, server_info *sinfo, unsigned int flags)
 
 	sinfo->pset_metadata_stale = 0;
 }
+
+/**
+ * @brief	Constructor for a server_psets array
+ *
+ * @param	nothing
+ *
+ * @return	server_psets *
+ * @retval	new server_psets object
+ * @retval	NULL for malloc error
+ */
+server_psets **
+new_server_psets(int num_psets)
+{
+	server_psets **ret = NULL;
+	int i;
+
+	ret = static_cast<server_psets **> (malloc((num_psets + 1) * sizeof(server_psets *)));
```

Comment:
We're C++ now, and this is completely new.  Consider using a vector.

Reply:
ok ill give it a shot